packages/python/plotly/plotly/figure_factory/_annotated_heatmap.py

```diff
@@ -264,34 +262,48 @@ def get_text_color(self):
             max_text_color = black
         return min_text_color, max_text_color
 
-    def get_z_mid(self):
+    def get_z_min_max(self):
         """
-        Get the mid value of z matrix
+        Get the min and max value of z matrix
 
-        :rtype (float) z_avg: average val from z matrix
+        :rtype (tuple): min and max val from z matrix
         """
         if np and isinstance(self.z, np.ndarray):
-            z_min = np.amin(self.z)
-            z_max = np.amax(self.z)
+            z_min = np.nanmin(self.z)
+            z_max = np.nanmax(self.z)
         else:
             z_min = min([v for row in self.z for v in row])
             z_max = max([v for row in self.z for v in row])
-        z_mid = (z_max + z_min) / 2
-        return z_mid
+        return z_min, z_max
 
-    def make_annotations(self):
+    def make_annotations(self, zmin=None, zmid=None, zmax=None):
         """
         Get annotations for each cell of the heatmap with graph_objs.Annotation
 
         :rtype (list[dict]) annotations: list of annotations for each cell of
             the heatmap
         """
         min_text_color, max_text_color = _AnnotatedHeatmap.get_text_color(self)
-        z_mid = _AnnotatedHeatmap.get_z_mid(self)
+        if zmin is None or zmax is None:
+            zmin, zmax = _AnnotatedHeatmap.get_z_min_max(self)
```

Comment:
here it seems like we overwrite both if either is missing? probably best not to do that I think.

Reply:
From what I understand from the doc https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Heatmap.html "zmax – Sets the upper bound of the color domain. Value should have the same units as in z and if set, zmin must be set as well." we can not pass only one of these 2 thats why i overwrite both

Reply:
This is true of the underlying heatmap trace, yes, but this figure-factory already does its own bounds-computation, so we may as well reuse it. I've already addressed this comment in #2921.